daemon/queue/UrlCoordinator.cpp

```diff
@@ -105,15 +103,13 @@ void UrlCoordinator::Run()
 			}
 		}
 
-		int sleepInterval = downloadStarted ? 0 : 100;
-		usleep(sleepInterval * 1000);
-
-		resetCounter += sleepInterval;
-		if (resetCounter >= 1000)
+		if (!downloadStarted && !IsStopped())
 		{
 			// this code should not be called too often, once per second is OK
 			ResetHangingDownloads();
-			resetCounter = 0;
+
+			UniqueLock lk(m_pauseMutex);
+			m_pauseCV.wait_for(lk, std::chrono::seconds(1), [&]{ return IsStopped() || m_hasMoreJobs; });
```

Comment:
Very high CPU usage (full load of one core) if the program was started with paused queue:

```shell
nzbget -s -P
```

This is caused by `m_hasMoreJobs` initialised to `true` in constructor; it's never recomputed above if download is paused.

Originally `m_hasMoreJobs` was used to inform main class (`nzbget.cpp`) that there is still job to do. It was set to `true` in constructor to make sure the main class doesn't stop preliminary before UrlCoordinator has checked the queue.